### PR TITLE
add ContextlessAnnotation

### PR DESCRIPTION
**Pull request by ericphanson**
when one only has a Samples object (and not a signal), and they run some algorithm over it that generates annotations, that algo doesn’t have the context of recording or start of the signal, and can only output things relative to the samples itself. That leads to the need for something like this `ContextlessSpanV1`.

More beacon-internal context: https://beacon-biosignals.slack.com/archives/CKS1Y5F5X/p1695907224398859

### CODE REVIEW

File: test/annotations.jl [hunk 1]
```diff
@@ -19,20 +19,20 @@ end
...
+        AnnotationV1(recording=recs[2], id=uuid4(), span=TimeSpan(15, 170)),               #= 6 =#
+        AnnotationV1(recording=recs[1], id=uuid4(), span=TimeSpan(43, 89)),               #= 7 =#
+        AnnotationV1(recording=recs[3], id=uuid4(), span=TimeSpan(0, 50)),               #= 8 =#
+        AnnotationV1(recording=recs[2], id=uuid4(), span=TimeSpan(2, 10)),               #= 9 =#
+        AnnotationV1(recording=recs[1], id=uuid4(), span=TimeSpan(111, 140)),               #= 10 =#
+        AnnotationV1(recording=recs[3], id=uuid4(), span=TimeSpan(60, 100)),               #= 11 =#
+        AnnotationV1(recording=recs[3], id=uuid4(), span=TimeSpan(23, 80)),               #= 12 =#
+        AnnotationV1(recording=recs[3], id=uuid4(), span=TimeSpan(100, 110)),               #= 13 =#
+        AnnotationV1(recording=recs[1], id=uuid4(), span=TimeSpan(200, 300))]               #= 14 =#
     merged = merge_overlapping_annotations(sources)
```
Comment (kleinschmidt) [inline test/annotations.jl:36]: these changes seem unrelated?

Comment (ericphanson) [reply test/annotations.jl:36]: formatter autoformatted, my bad. Will fix

**Comment by ericphanson** [comment]
On slack we decided the `id` isn't necessary here, and this can just be a `Vector{TimeSpan}`, or some other table (if other columns are needed too for some purpose), but we don't need a `ContextlessAnnotation` object.
